src/main/java/com/programmers/smrtstore/domain/auth/application/AuthService.java

```diff
@@ -0,0 +1,55 @@
+package com.programmers.smrtstore.domain.auth.application;
+
+import com.programmers.smrtstore.core.properties.ErrorCode;
+import com.programmers.smrtstore.domain.auth.application.dto.req.SignUpRequest;
+import com.programmers.smrtstore.domain.auth.application.dto.res.LoginResponse;
+import com.programmers.smrtstore.domain.auth.application.dto.res.SignUpResponse;
+import com.programmers.smrtstore.domain.auth.domain.entity.Auth;
+import com.programmers.smrtstore.domain.auth.exception.AuthException;
+import com.programmers.smrtstore.domain.auth.infrastructure.AuthRepository;
+import com.programmers.smrtstore.domain.user.domain.entity.User;
+import com.programmers.smrtstore.domain.user.infrastructure.UserRepository;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class AuthService {
+
+    private final PasswordEncoder passwordEncoder;
+    private final AuthRepository authRepository;
+    private final UserRepository userRepository;
+
+    @Transactional(readOnly = true)
+    public LoginResponse login(String username, String password) {
+        Auth auth = authRepository.findByUsername(username)
+            .orElseThrow(() -> new AuthException(ErrorCode.USER_NOT_FOUND));
+        auth.verifyPassword(passwordEncoder, password);
+        return LoginResponse.from(auth);
+    }
+
+    public SignUpResponse signUp(@Valid SignUpRequest request) {
+        authRepository.findByUsername(request.getUsername()).ifPresent(auth -> {
```

Comment:
하단에 동일한 로직의 checkDuplicateUsername 메서드를 만들어놓고 여기서는 사용하지 않는 이유가 따로 있나용?

Reply:
그러게요!!

Reply:
`@Transactional`이 적용된 메서드 내부에서 `@Transactional`이 적용된 메서드를 활용하면 아래와 같은 문제점이 있습니다!

> A method annotated with Spring’s @Async or @Transactional annotations will not work as expected if invoked directly from within its class.
This is because Spring generates a proxy class with wrapper code to manage the method’s asynchronicity (@Async) or to handle the transaction (@Transactional). However, when called using this, the proxy instance is bypassed, and the method is invoked directly without the required wrapper code.
Spring의 @Async 또는 @Transactional 어노테이션으로 어노테이션된 메서드는 해당 클래스 내에서 직접 호출하면 예상대로 작동하지 않습니다.
이는 Spring이 메서드의 비동기성(@Async)을 관리하거나 트랜잭션(@Transactional)을 처리하기 위해 래퍼 코드가 포함된 프록시 클래스를 생성하기 때문입니다. 그러나 이를 사용하여 호출하면 프록시 인스턴스가 우회되고 필요한 래퍼 코드 없이 메서드가 직접 호출됩니다.
from SonarLint java:S6809

이를 해결하고자 분리해서 사용했습니다 :D

Reply:
생각해보니 `checkDuplicateUsername` 메서드가 readOnly라서 트렌젝션의 영향을 받지 않을 것 같네요! 수정하도록 하겠습니다 :D 👍🏻